### PR TITLE
Flake8 pyi

### PR DESCRIPTION
**Pull request by JonathanPlasse**
### Purpose of the pull request
Update the code generation not to generate some flake8-pyi errors.

### CODE REVIEW

**Review by nutti — COMMENTED** [review]
@JonathanPlasse 

Does this PR relate to https://github.com/nutti/fake-bpy-module/issues/161#issue-2045306391 ?

> Add Ruff and enable linter rules to avoid regression we encountered

If this PR supports the above item, could you add the new linter test for the generated code in CI pipeline?

**Comment by JonathanPlasse** [comment]
> could you add the new linter test for the generated code in CI pipeline?

I added flake8-pyi to run in the CI.
I ignore 5 rules `PYI011`, `PYI014`, `PYI018`, `PYI021`, `PYI054`.
They need to be ignored otherwise the CI fails (c.f. https://github.com/nutti/fake-bpy-module/actions/runs/10405883813/job/28817692685`)
Screenshot from 2024-08-15 17-16-18

These ignored rules could be addressed in follow-up PRs.

File: .github/workflows/fake-bpy-module-ci.yml [hunk 1]
```diff
@@ -134,6 +134,9 @@ jobs:
           name: fake_bpy_modules_${{ matrix.blender_version}}_raw_${{ needs.set_versions.outputs.file_version }}
           path: "raw_modules/fake_bpy_module*"
 
+      - name: Run Ruff flake8-pyi checks
```
Comment (nutti) [inline .github/workflows/fake-bpy-module-ci.yml:137]: The generated code can be found at the path `raw_modules/fake_bpy_module*` as zip files.
Extract them to the some temporary directories and lint them is better which tests the artifacts itself.

File: tools/pip_package/build_pip_package.sh [hunk 2]
```diff
@@ -195,4 +195,3 @@ fi
 
 # clean up
 cd "${CURRENT_DIR}"
-rm -rf "${tmp_dir}"
```
Comment (nutti) [inline tools/pip_package/build_pip_package.sh:198]: If you lint the artifact, this change is not required any more.

**Review by nutti — CHANGES_REQUESTED** [review]

File: tools/pip_package/run_ruff_flake8_pyi.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    echo "No Blender version provided. Usage: $0 <blender_version>"
+    exit 1
+fi
+
+BLENDER_VERSION=$1
+ZIP_FILE="fake_bpy_modules_${BLENDER_VERSION}-*.zip"
+EXTRACT_DIR="extracted_modules"
+
+# Find and unzip the correct file
+ZIP_PATH=$(find . -name "$ZIP_FILE" -print -quit)
```
Comment (nutti) [inline tools/pip_package/run_ruff_flake8_pyi.sh:15]:
```suggestion
ZIP_PATH=$(find . -name "${ZIP_FILE}" -print -quit)
```

File: tools/pip_package/run_ruff_flake8_pyi.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+if [ -z "$ZIP_PATH" ]; then
+    echo "No matching zip file found for Blender version $BLENDER_VERSION"
+    exit 1
+fi
```
Comment (nutti) [inline tools/pip_package/run_ruff_flake8_pyi.sh:19]:
```suggestion
if [ -z "${ZIP_PATH}" ]; then
    echo "No matching zip file found for Blender version ${BLENDER_VERSION}"
    exit 1
fi
```

File: tools/pip_package/run_ruff_flake8_pyi.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+mkdir -p "$EXTRACT_DIR"
+unzip -q "$ZIP_PATH" -d "$EXTRACT_DIR"
```
Comment (nutti) [inline tools/pip_package/run_ruff_flake8_pyi.sh:22]:
```suggestion
mkdir -p "${EXTRACT_DIR}"
unzip -q "${ZIP_PATH}" -d "{$EXTRACT_DIR}"
```

File: tools/pip_package/run_ruff_flake8_pyi.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+# Run Ruff flake8-pyi checks
+ruff check "$EXTRACT_DIR" --select=PYI --ignore=PYI011,PYI014,PYI018,PYI021,PYI054
```
Comment (nutti) [inline tools/pip_package/run_ruff_flake8_pyi.sh:25]:
```suggestion
ruff check "${EXTRACT_DIR}" --select=PYI --ignore=PYI011,PYI014,PYI018,PYI021,PYI054
```

File: tools/pip_package/run_ruff_flake8_pyi.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# Clean up
+rm -rf "$EXTRACT_DIR"
```
Comment (nutti) [inline tools/pip_package/run_ruff_flake8_pyi.sh:28]:
```suggestion
rm -rf "${EXTRACT_DIR}"
```

**Review by nutti — CHANGES_REQUESTED** [review]
Thank you for the changes.
I left some trivial comments.
Could you apply them?

**Review by nutti — APPROVED** [review]
@JonathanPlasse 
From the discussion on Discord, I applied my suggested changes.

**Pull request merged**